File: 1444-number-of-steps-to-reduce-a-number-to-zero/number-of-steps-to-reduce-a-number-to-zero.py
class Solution:
    def numberOfSteps(self, num: int) -> int:
        count = 0

        def reducer(num, count):
            if num == 0:
                return count
            
            if num % 2 == 0:
                count += 1
                return reducer(num/2,count)
            else:
                count += 1
                return reducer(num - 1,count)

        return reducer(num,count)

File: 1444-number-of-steps-to-reduce-a-number-to-zero/number-of-steps-to-reduce-a-number-to-zero.java
class Solution {
    public int numberOfSteps(int num) {
        return reducer(num,0);
    }
    static int reducer(int num,int count){
        if (num == 0)return count;

        if (num % 2 == 0) return reducer(num/2,++count);
        else return reducer(--num,++count);
    }
}